app/shared/components/BigButton.svelte

```diff
@@ -0,0 +1,64 @@
+<script type="ts" context="module">
+
+export function generateIcon(preferredJourneyMode:string){
+  if(preferredJourneyMode === "fastest") return "north"
+  if(preferredJourneyMode === "least-changes") return "switch_access_shortcut_add"
+  if(preferredJourneyMode === "least-walking") return "directions_walk"
+
+  return ""
+}
+
+</script>
+
+<script type="ts">
+
+  import { createEventDispatcher } from "svelte";
+  import { PreferredJourneyMode } from "~/types";
+
+  export let icon = ""
+  $: hasIcon = icon !== ""
```

Comment:
actually ok. but this var is only used in one place. The bool could be included there